proxystore/connectors/dim/margo.py

```diff
@@ -84,32 +87,44 @@ class MargoConnector:
 
     def __init__(
         self,
-        interface: str,
         port: int,
         protocol: Protocol | str,
+        addr: str | None = None,
+        interface: str | None = None,
         timeout: float = 1,
     ) -> None:
         # Py-Mochi-Margo is not a required dependency and requires the user
         # to install it themselves.
         if pymargo_import_error is not None:  # pragma: no cover
             raise pymargo_import_error
 
-        self.interface = interface
+        self.host: str = ''
```

Comment:
Nit: the switch from `addr` to `host` seems unnecessary. Let's stick with `address` to make it clear these are IP addresses and not hostnames.